nomos-services/chain/chain-leader/src/leadership.rs

```diff
@@ -40,37 +35,29 @@ impl Leader {
     )]
     pub async fn build_proof_for(
         &self,
-        aged_tree: &UtxoTree,
         latest_tree: &UtxoTree,
         epoch_state: &EpochState,
         slot: Slot,
     ) -> Option<Groth16LeaderProof> {
         for utxo in &self.utxos {
-            let Some(_aged_witness) = aged_tree.witness(&utxo.id()) else {
-                continue;
-            };
-            let Some(_latest_witness) = latest_tree.witness(&utxo.id()) else {
+            let Some(public_inputs) = public_inputs_for_utxo(utxo, epoch_state, slot, latest_tree)
+            else {
                 continue;
             };
 
-            let note_id: Fr = BigUint::from(1u8).into(); // placeholder for note ID, replace after mantle notes format update
-            let public_inputs = LeaderPublic::new(
-                aged_tree.root(),
-                latest_tree.root(),
-                epoch_state.nonce,
-                slot.into(),
-                epoch_state.total_stake(),
-            );
+            let note_id = utxo.note.id();
```

Comment:
Note Id comes from the Utxo

```suggestion
            let note_id = utxo.id();
```